File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/08 Check Palindrome/palindrome.java
class Solution {
    // Helper function to check palindrome recursively
    private boolean helper(String s, int start, int end) {
        if (start >= end) {
            return true;
        }
        if (s.charAt(start) == s.charAt(end)) {
            return helper(s, start + 1, end - 1);
        }
        return false;
    }

    public boolean isPalindrome(String s) {
        // Convert string to lowercase
        s = s.toLowerCase();

        // Remove non-alphanumeric characters
        StringBuilder filtered = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                filtered.append(c);
            }
        }

        // Call the helper function
        String cleanString = filtered.toString();
        return helper(cleanString, 0, cleanString.length() - 1);
    }
}

File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/08 Check Palindrome/palindrome.py
class Solution(object):
    def helper(self, s, start, end):
        bool_store = False
        if(start>=end):
            bool_store = True
            return bool_store
        if s[start] == s[end]:
            bool_store = self.helper(s , start+1, end-1)
        return bool_store

    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s=s.lower()
        s = list((i for i in s if i.isalnum()))
        s = "".join(s)
        boo = self.helper(s, 0, len(s)-1)
        return boo
        